src/Discord/Internal/Types/ApplicationInfo.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-- | Data structures pertaining to Discord User
+module Discord.Internal.Types.ApplicationInfo where
+
+import Data.Aeson
+import Data.Text (Text)
+import qualified Data.Text as T
+import Discord.Internal.Types.Prelude
+import Data.Time (UTCTime)
+
+
+-- | Structure containing partial information about an Application
+data FullApplication = FullApplication
+  { fullApplicationID :: ApplicationId
+  , fullApplicationName :: T.Text
+  , fullApplicationFlags :: Int
+  } deriving (Show, Eq, Read)
+
+instance FromJSON FullApplication where
+  parseJSON = withObject "FullApplication" $ \o ->
+    FullApplication <$> o .: "id"
+                    <*> o .: "name"
+                    <*> o .: "flags"
+
```

Comment:
Can be filled in later with all the fields https://discord.com/developers/docs/resources/application

Reply:
Could you make a ticket for this?